lib/web/userpreferences.go

```diff
@@ -243,3 +262,16 @@ func accessGraphPreferencesResponse(resp *userpreferencesv1.AccessGraphUserPrefe
 		HasBeenRedirected: resp.HasBeenRedirected,
 	}
 }
+
+// accessGraphPreferencesResponse creates a JSON response for the access graph preferences.
+func discoverResourcePreferenceResponse(resp *userpreferencesv1.DiscoverResourcePreferences) DiscoverResourcePreferencesResponse {
+	if resp == nil || resp.DiscoverGuidePreferences == nil {
```

Comment:
nit: use `GetX` where possible as it adds safety from nil pointer
```suggestion
	if resp == nil || resp.GetDiscoverGuidePreferences == nil {
```